.github/workflows/go.yml

```diff
@@ -1,25 +1,129 @@
-name: Go
+name: Build, Test and Release
 
 on:
   push:
-    branches: [ main ]
+    branches:
+      - main
+    tags:
+      - v*.*.*
   pull_request:
-    branches: [ main ]
+    branches:
+      - main
 
 jobs:
 
   build:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+
+    env:
+      GO_LDFLAGS: "-w -s"
+      GO11MODULE: "on"
+      GINKGO_FLAGS: "--randomizeAllSpecs --randomizeSuites --failOnPending --trace --race --progress --compilers=2 --nodes=2"
+      VERSION: ${{ startsWith(github.ref, 'refs/tags/v') && github.ref_name || format('gh-action-{0}', github.sha) }}
+
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # renovate: tag=v2
+
+      - name: Setup Cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/go/pkg
+          key: ${{ runner.os }}-build-${{ runner.arch }}
+
+      - name: Set up Go
+        uses: actions/setup-go@331ce1d993939866bb63c32c6cbbfd48fa76fc57 # renovate: tag=v2
+        with:
+          go-version: 1.17
+
+      - name: Build Binary
+        run: |
+          make generate
+          if ! git diff --exit-code --name-only; then
+            git diff
+            echo 'Run "make generate" and add generated changes to this PR'
+            exit 1
+          fi
+          # Build kubecfg
+          make all
+
+      - name: Info Binary
+        run: |
+          #ldd ./kubecfg || otool -L ./kubecfg # What's the purpose of this ? it currently fail since we don't produce a dynamic executable
```

Comment:
on macos it is.

```console
$ otool -L ./kubecfg
./kubecfg:
	/usr/lib/libSystem.B.dylib (compatibility version 0.0.0, current version 0.0.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 0.0.0, current version 0.0.0)
	/System/Library/Frameworks/Security.framework/Versions/A/Security (compatibility version 0.0.0, current version 0.0.0)
```

anyway it may be useful to show whether we indeed built a fully static executable or not.

Reply:
ok, so what is the purpose ? to _succed_ if is a static file ? 

it was failing for me . mind if i re-enable it after this PR is merged ?

Reply:
nah, let's just drop it

Reply:
I mean, if somebody wants to see that info, they can download the artifact and run the tool;

the only point of this info is to watch the mac info if you don't have a mac; it doesn't really matter I think.

Reply:
:+1: